bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -198,10 +200,19 @@ public void stateUpdated(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
+                // event.item contains data from the item including its state (in event.item.state)
                 event.item = EnrichedItemDTOMapper.map(item, false, null, null);
 
-                // adjust the state according to the widget type
-                event.item.state = itemUIRegistry.getState(w).toFullString();
+                // event.state is an adjustment of the item state to the widget type.
+                // In case this state is big and identical to the item state, an indirection to the
+                // item state is provided by setting its value to "#item.state#" in order to avoid
+                // duplicating in the event message a big amount of data.
+                String state = itemUIRegistry.getState(w).toFullString();
+                if ((state.length() > 128) && state.equals(event.item.state)) {
```

Comment:
The client does need to handle `REDIRECT_TO_ITEM_STATE`, so why using the magic number 128?
If the client check if `event.state` is equal to `REDIRECT_TO_ITEM_STATE` and use `event.item.state` wouldn't already a length greater then `REDIRECT_TO_ITEM_STATE.length()` make sense?

Reply:
My intention was to reduce the use of redirection in most of the cases (for reasonable size), for a maybe better readability and for a working in most cases for UI client that even forget to consider the redirection.
Your proposal is more for the optimal size.
I am ok to change that.